cmd/serverless/main.go

```diff
@@ -119,7 +119,13 @@ func runAgent() {
 
 	debug.OutputDatadogEnvVariablesForDebugging()
 
-	if !apikey.HasAPIKey() {
+	config.Datadog.SetConfigFile(datadogConfigPath)
+	// Load datadog.yaml file into the config, so that metricAgent can pick these configurations
+	if _, err := config.LoadWithoutSecret(); err != nil {
+		log.Errorf("Error happened when loading configuration from datadog.yaml for metric agent: %s", err)
+	}
+
+	if err := apikey.HandleEnv(); err != nil {
```

Comment:
I removed `apikey.HasAPIKey()` because it seemed redundant to check `config.Datadog.IsSet("api_key")` in that function and then again in `apikey.HandleEnv()`